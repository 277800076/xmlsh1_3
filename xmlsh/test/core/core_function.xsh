# Test of function declarations

# Function read in source mode

unset _A
foo()
{
  echo foo: args are $*
  _A=foo
}

function bar()
{
  echo bar: args are $*
  _A=bar
}

unset _A
foo arg1 arg2 arg3
echo after foo _A is $_A
unset _A
bar arg1 arg2
echo after bar _A is $_A

# Test that functions don't thrash $*

set A B C
bar arg1 arg2 
[  $# -eq 3 ] || { echo FAIL : expected to preserve args ; }
foo arg1 arg2
[  $# -eq 3 ] || { echo FAIL : expected to preserve args ; }



