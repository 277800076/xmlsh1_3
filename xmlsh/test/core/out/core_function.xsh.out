foo: args are arg1 arg2 arg3
after foo _A is foo
bar: args are arg1 arg2
after bar _A is
bar: args are arg1 arg2
foo: args are arg1 arg2
